cmd/microservice-sui-worker/transaction.go

```diff
@@ -0,0 +1,142 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+
+	"github.com/block-vision/sui-go-sdk/models"
+	"github.com/block-vision/sui-go-sdk/signer"
+	suiSdk "github.com/block-vision/sui-go-sdk/sui"
+	"github.com/fluidity-money/fluidity-app/lib/log"
+	"github.com/fluidity-money/fluidity-app/lib/types/sui"
+	worker_types "github.com/fluidity-money/fluidity-app/lib/types/worker"
+)
+
+// TODO utility in EthereumSpooledRewards has to be the address of the utility token
+// createPayoutTransaction to call distribute_yield on the fluid token contract
+func createPayoutTransaction(client suiSdk.ISuiAPI, fluidToken, baseToken sui.SuiToken, rewards worker_types.EthereumSpooledRewards, suiPayoutArgs payoutArgs, workerAddress string) (models.TxnMetaData, error) {
+	var (
+		prizePoolVault = suiPayoutArgs.PrizePoolVault
+		scallopVersion = suiPayoutArgs.ScallopVersion
+		scallopMarket  = suiPayoutArgs.ScallopMarket
+		clock          = suiPayoutArgs.Clock
+	)
+
+	var winners []string
+	var amounts []uint64
+
+	for _, r := range rewards.Rewards {
+		for address, win := range r {
+			winners = append(winners, address.String())
+			amounts = append(amounts, win.Uint64())
+		}
+	}
+
+	// manually find the largest SUI token to use for gas payment
+	// assumes that the worker has at least one coin with enough balance to pay gas
+	// and doesn't perform gas smashing
+	coinsResponse, err := client.SuiXGetCoins(context.Background(), models.SuiXGetCoinsRequest{Owner: workerAddress})
```

Comment:
How will this behave if it's set to context background?

Reply:
We need to figure out a strategy if there's too much coin fragmentation for the gas coin spend